pkg/cmd/admin/image/verify-signature.go

```diff
@@ -0,0 +1,368 @@
+		panic(fmt.Sprintf("Internal inconsistency: policyconfiguration.DockerReferenceIdentity returned %#v, %v", res, err))
+	}
+	return res
+}
+
+// PolicyConfigurationNamespaces returns a list of other policy configuration namespaces to search
+// for if explicit configuration for PolicyConfigurationIdentity() is not set.  The list will be processed
+// in order, terminating on first match, and an implicit "" is always checked at the end.
+// It is STRONGLY recommended for the first element, if any, to be a prefix of PolicyConfigurationIdentity(),
+// and each following element to be a prefix of the element preceding it.
+func (ref fakeDockerReference) PolicyConfigurationNamespaces() []string {
+	return policyconfiguration.DockerReferenceNamespaces(ref.ref)
+}
+
+func (ref fakeDockerReference) NewImage(ctx *sigtypes.SystemContext) (sigtypes.Image, error) {
+	panic("Unimplemented")
+}
+func (ref fakeDockerReference) NewImageSource(ctx *sigtypes.SystemContext, requestedManifestMIMETypes []string) (sigtypes.ImageSource, error) {
+	panic("Unimplemented")
+}
+func (ref fakeDockerReference) NewImageDestination(ctx *sigtypes.SystemContext) (sigtypes.ImageDestination, error) {
+	panic("Unimplemented")
+}
+func (ref fakeDockerReference) DeleteImage(ctx *sigtypes.SystemContext) error {
+	panic("Unimplemented")
+}
+
+// unparsedImage implements sigtypes.UnparsedImage, to allow evaluating the signature policy
+// against an image without having to make it pullable by containers/image
+type unparsedImage struct {
+	ref       sigtypes.ImageReference
+	manifest  []byte
+	signature []byte
+}
+
+func newUnparsedImage(expectedIdentity string, img *imageapi.Image, signature []byte) (sigtypes.UnparsedImage, error) {
+	ref, err := parseDockerReference("//" + expectedIdentity)
+	if err != nil {
+		return nil, fmt.Errorf("Invalid --expected-identity: %v", err)
+	}
+	return &unparsedImage{ref: ref, manifest: []byte(img.DockerImageManifest), signature: signature}, nil
+}
+
+// Reference returns the reference used to set up this source, _as specified by the user_
+// (not as the image itself, or its underlying storage, claims).  This can be used e.g. to determine which public keys are trusted for this image.
+func (ui *unparsedImage) Reference() sigtypes.ImageReference {
+	return ui.ref
+}
+
+// Close removes resources associated with an initialized UnparsedImage, if any.
+func (ui *unparsedImage) Close() error {
+	return nil
+}
+
+// Manifest is like ImageSource.GetManifest, but the result is cached; it is OK to call this however often you need.
+func (ui *unparsedImage) Manifest() ([]byte, string, error) {
+	return ui.manifest, "", nil
+}
+
+// Signatures is like ImageSource.GetSignatures, but the result is cached; it is OK to call this however often you need.
+func (ui *unparsedImage) Signatures() ([][]byte, error) {
+	return [][]byte{ui.signature}, nil
+}
+
+func (o *VerifyImageSignatureOptions) Run() error {
```

Comment:
@soltysh i don't get why the validate must be after complete... in my case, the complete will send a request to api to get the current username, deal with clients, etc... and then the command will die because you specify invalid combo of options...

in this case, i found more logically that I first validate the options/flags provided and then move to completion of the options.